[GR-45874] Update GitHub Issue Template for Improved User Issue Reporting

fernando-valdez opened the pull request:
### Summary
This PR updates the GitHub issue template to help users file new issues more effectively. The changes aim to streamline the issue reporting process, ensuring that users provide all necessary information for quicker and more accurate troubleshooting.

### Changes
- Added new sections for environment details, steps to reproduce, and expected behavior.
- Included guidelines for providing clear and concise issue titles.
- Added placeholders for logs and screenshots to better illustrate issues.
- Updated formatting for improved readability and user guidance.

### Benefits
- Encourages users to provide comprehensive details, reducing back-and-forth communication.
- Helps maintainers and contributors understand and reproduce issues faster.
- Improves the overall efficiency of issue triage and resolution.

### Checklist
- [x] Reviewed the new template for completeness and clarity.
- [x] Tested the template with sample issues to ensure usability.

woess committed:
Remove constant memory buffer assumption and simplify ByteArrayWasmMemory.

woess committed:
Check that memory 0 size >= initial size at the start of a wasm function.

This allows hoisting bounds checks for memory accesses that are guaranteed to stay within the initial bounds.

woess committed:
Only perform initial memory 0 bounds check if the wasm function uses memory 0.

woess committed:
Work around for nullary benchmarkTeardownEach() in photon benchmark.

woess committed:
Fix TraceTransferToInterpreter in WasmFunctionNode.profileCondition.

fernando-valdez committed:
updates issue template files

fernando-valdez committed:
Merge branch 'oracle:master' into fv/GR-45874/update-issues-template

fernando-valdez committed:
Merge branch 'fv/GR-45874/update-issues-template'
Updated issues template

woess committed:
Try to read WebAssembly.Module bytes via Buffer Interop.

jirkamarsik committed:
Implement WASI's fd_renumber

jirkamarsik committed:
Fix type error in WasiUnsupportedFunctionNode

jirkamarsik committed:
Use fdRightsInheriting when checking rights of derived fds

jirkamarsik committed:
Move fd_renumber logic to WasiFdRenumberNode

jirkamarsik committed:
Implement WASI's clock_res_get

vjovanov commented:
Please rebase it to master so there are no merge commits.

vjovanov commented on .github/ISSUE_TEMPLATE/1_3_native_image_error_report.yml:
```diff
@@ -0,0 +1,59 @@
+  - type: textarea
+    id: error_report_file
+    attributes:
+      label: Share the `svm_err_b_*.md` File
```
we can use the same text as below

vjovanov commented on .github/ISSUE_TEMPLATE/1_3_native_image_error_report.yml:
```diff
@@ -0,0 +1,59 @@
+name: 🚨 Share Native Image Error Report
```
Maybe call this: `Share Native Image Build-Time Crash Report`? @fniephaus

vjovanov replied:
We could also use a different icon, e.g., 💥

fernando-valdez replied:
The 🚨 icon indicates a native image issue. I updated a few more icons to match the context of the other components too.

fernando-valdez replied:
Change included. And code updated

vjovanov commented:
If the crash report can not be fused with build-time error, I think we are good for now. We will see how the fields work in practice and adjust.

jirkamarsik committed:
Implement WASI's fd_tell, check offset in fd_seek

jirkamarsik committed:
Implement WASI's fd_datasync

jirkamarsik committed:
Implement WASI's fd_pread and fd_pwrite

fernando-valdez committed:
Merge branch 'oracle:master' into fv/GR-45874/update-issues-template

jirkamarsik committed:
Avoid using SeekableByteChannel#position on files opened in append mode

jirkamarsik committed:
Implement WASI's fd_sync

jirkamarsik committed:
Implement WASI's fd_filestat_set_size

jirkamarsik committed:
Implement WASI's fd_fdstat_set_rights

jirkamarsik committed:
Implement WASI's fd_readdir

jirkamarsik committed:
Handle fd == to in fd_renumber(fd, to)

mur47x111 committed:
Update SyncPort info.

jirkamarsik committed:
Fix rights checking in WASI's path_open

mur47x111 committed:
Avoid getHostBackend when instantiating UnimplementedGraalIntrinsics.

jirkamarsik committed:
Simplify error code checks in FdUtils methods

jirkamarsik committed:
Improve naming of WASI implementation methods

cstancu committed:
Make BootStrapMethodHandler more general.

Allows it to be used by the ClassInitializerBytecodeParser used by the AutomaticUnsafeTransformationSupport.

fernando-valdez committed:
Merge branch 'oracle:master' into master

fernando-valdez committed:
Change the name of the error report template

fernando-valdez committed:
Merge branch 'fv/GR-45874/update-issues-template'

fernando-valdez committed:
add compiler issue report form

vjovanov commented on .github/ISSUE_TEMPLATE/1_3_native_image_error_report.yml:
```diff
@@ -0,0 +1,59 @@
+name: 💥 Share Native Image Build-Time Crash Report (svm_err_b_*.md)
+description: Share an error report for GraalVM's native image using the generated `svm_err_b_*.md` file
```
`Share a crash report`

vjovanov commented on .github/ISSUE_TEMPLATE/1_3_native_image_error_report.yml:
```diff
@@ -0,0 +1,59 @@
+name: 💥 Share Native Image Build-Time Crash Report (svm_err_b_*.md)
```
I would change the file name to match crash.

vjovanov commented on .github/ISSUE_TEMPLATE/1_3_native_image_error_report.yml:
```diff
@@ -0,0 +1,59 @@
+name: 💥 Share Native Image Build-Time Crash Report (svm_err_b_*.md)
+description: Share an error report for GraalVM's native image using the generated `svm_err_b_*.md` file
+title: "[Native Image] Error Report"
```
Crash Report

vjovanov commented on .github/ISSUE_TEMPLATE/1_3_native_image_error_report.yml:
```diff
@@ -0,0 +1,59 @@
+    id: describe_issue
+    attributes:
+      label: Describe the Issue
+      description: "Describe the error you encountered with GraalVM's native image. Provide a clear and concise description of what happened and what you were trying to achieve."
```
Native Image

vjovanov commented on .github/ISSUE_TEMPLATE/1_3_native_image_error_report.yml:
```diff
@@ -0,0 +1,59 @@
+      value: |
+        ### How to Find and Share the `svm_err_b_*.md` File
+
+        When a native image build fails, GraalVM generates a detailed error report file with a name like `svm_err_b_<timestamp>_pid<pid>.md`. This file contains valuable information for diagnosing the issue.
```
pid is mentioned twice.
fails -> fails fatally

mur47x111 committed:
Update SyncPort info.

fernando-valdez committed:
update the crash report template

vjovanov commented:
Please squash all commits and rebase to master. There should be no merge commits.

vjovanov commented:
There needs to be a single commit that is merged. Please assure that we don't merge 36 commits.
